lib/rust/ensogl/core/src/application/frp.rs

```diff
@@ -506,3 +506,536 @@ macro_rules! define_endpoints_emit_alias {
+                        let command_map = Rc::new(RefCell::new(command_map));
+
+                        let _phantom_type_args = default();
+                        Self {status_map,command_map,$($out_field),*,_phantom_type_args }
+                    }
+                }
+
+
+                // === Combined ===
+
+                #[derive(Debug, CloneRef, Clone)]
+                pub struct Combined $(<$($param $(:$($constraints)*)?),*>)? {
+                    data: Rc<CombinedData$(<$($param),*>)?,> // deref
+                }
+
+                #[allow(unused_parens)]
+                impl $(<$($param $(:$($constraints)*)?),*>)? Combined $(<$($param),*>)? {
+                    pub fn new(input:&InputData$(<$($param),*>)?, output:&OutputData$(<$($param),*>)?) -> Self {
+                        let data = Rc::new(CombinedData::new(input,output));
+                        Self { data }
+                    }
+
+                    $($crate::define_endpoints_emit_alias!{$in_field ($($in_field_type)*)})*
+
+                }
+
+                impl $(<$($param $(:$($constraints)*)?),*>)? Deref for Combined $(<$($param),*>)? {
+                    type Target = CombinedData $(<$($param),*>)?;
+                    fn deref(&self) -> &Self::Target {
+                        &self.data
+                    }
+                }
+
+                #[allow(unused_parens)]
+                #[derive(Debug)]
+                pub struct CombinedData $(<$($param $(:$($constraints)*)?),*>)? {
+                    $( $(#[doc=$($in_doc)*])* pub $in_field : $crate::frp::Any<($($in_field_type)*)>,)*
+
+                    $($(#[doc=$($out_doc)*])*
+                        pub $out_field  : $crate::frp::Sampler<($($out_field_type)*)>,
+                    )*
+                }
+
+                impl $(<$($param $(:$($constraints)*)?),*>)? CombinedData $(<$($param),*>)? {
+                    pub fn new(input:&InputData$(<$($param),*>)?, output:&OutputData$(<$($param),*>)?) -> Self {
+                        $(let $in_field = input.$in_field.clone_ref();)*
+                        $(let $out_field = output.$out_field.clone_ref();)*
+
+                        Self{ $($in_field,)* $($out_field),* }
+                    }
+                }
+
+            }
+
+
+
+            // ===============
+            // === Private ===
+            // ===============
+
+            // No Clone. We do not want `network` to be cloned easily in the future.
+            #[derive(Debug)]
+            pub struct Private $(<$($param $(:$($constraints)*)?),*>)? {
+                pub network: $crate::frp::Network,
+                pub input: private::Input$(<$($param),*>)?,
+                pub output: private::Output$(<$($param),*>)?,
+            }
+
+
+            impl $(<$($param $(:$($constraints)*)?),*>)?  Private $(<$($param),*>)? {
+                pub fn new() -> Self {
+                    unimplemented!()
```

Comment:
is this intentional?

Reply:
No, this was left over and is now removed.